wgpu-hal/src/gles/mod.rs

```diff
@@ -251,6 +251,7 @@ struct AdapterShared {
     next_shader_id: AtomicU32,
     program_cache: Mutex<ProgramCache>,
     es: bool,
+    max_samples: i32,
```

Comment:
let's call this `max_msaa_samples` so it's actually clear what is meant for the next poor soul stumbling in here :). Also a note on why this is cached can't hurt:

```suggestion
    /// Result of `gl.get_parameter_i32(glow::MAX_SAMPLES)`.
    /// Cached here so it doesn't need to be queried every time texture format capabilities are requested.
    /// (this has been shown to be a significant enough overhead)
    max_msaa_samples: i32,
```

Reply:
Changed, thanks for the suggestion!